### PR TITLE
Reduce Vale linter errors on valid technical terms in Quarkus docs

### PR DESCRIPTION
**Pull request by michelle-purcell**
This PR adds the following exceptions to Quarkus grammar and style rules to prevent false positive errors in the vale linter report:

- gRPC
- xDS
- customizer
- getters
- setters
- metamodel

and updated the special terms detection rule to change the spelling error message contributors currently see on  "repo" to guidance to use "repository" instead.

### CODE REVIEW

**Comment by michelle-purcell** [comment]
@cescoffier - This PR should reduce the churn that you experienced in the Vale linter report in #30203 plus fixes for some other commonly used Java terms that flagged in other reports. Thanks for raising :+1:

**Review by cescoffier — APPROVED** [review]

**Comment by michelle-purcell** [comment]
Added guidance to use "repository" instead of "repo".

@ebullient - `dev` is another term frequently flagged by the rules as a spelling warning, which is not entirely correct. 

What do you think about the use of 'dev' throughout the docs? 

We could change the spelling warning to a term suggestion, with a message that's more specific like:
`Do not use as a short-hand for development or developer.`.

**Comment by ebullient** [comment]
> What do you think about the use of 'dev' throughout the docs?
> 
> We could change the spelling warning to a term suggestion, with a message that's more specific like: `Do not use as a shorthand for development or developer.`.

I get tangled up by our use of "dev mode" .. it is probably technically/grammatically "development mode", but our talks and blog posts stick with the short form. Is it possible to warn about use of `dev` unless the word mode comes next?

@maxandersen .. wdyt?

**Comment by michelle-purcell** [comment]
> I get tangled up by our use of "dev mode" .. it is probably technically/grammatically "development mode", but our talks and blog posts stick with the short form. Is it possible to warn about use of `dev` unless the word mode comes next?
> 

Yes. We can use a regular expression to remove the spelling warning on dev by adding an exception, and then add a new rule to detect instances of the string 'dev' that are not proceeded by the word 'node'.

I pushed a new commit to this PR that does exactly this. Here's a screen shot of the results when tested with 'dev node' 'dev experience' and  'dev mode'. Note that no correction is needed for 'dev mode' below.

image

Is this OK?

File: .gitignore [hunk 1]
```diff
@@ -41,3 +41,4 @@ nb-configuration.xml
 .sdkmanrc
 .envrc
 .jekyll-cache
+vale.ini
```
Comment (rolfedh) [inline .gitignore:44]:
```suggestion
.vale.ini
```
Do you want to include the dot prefix in the file name?

**Comment by maxandersen** [comment]
Should also accept "quarkus dev" but maybe that's ignored for things in verbatim?

File: .gitignore (hunk 1 above)
Comment (ebullient) [inline .gitignore:44]: vale.ini should not be ignored unless/until we move things to .vale.ini, which @maxandersen may change in #29913

**Review by ebullient — CHANGES_REQUESTED** [review]

**Comment by ebullient** [comment]
> Should also accept "quarkus dev" but maybe that's ignored for things in verbatim?

yes.. ignored for things in verbatim / code blocks

File: .gitignore (hunk 1 above)
Comment (michelle-purcell) [reply .gitignore]: @rolfedh - oops. Didn't mean to commit this config change. I've reversed this bit of the commit and added the config to my local global git ignore file. Thanks for flagging though :+1: 

In the Quarkus implementation of Vale, the config file is without the preceding dot (.). I use it in the root of my local IDE for testing the rule tweaks.

image

Comment (michelle-purcell) [reply .gitignore]: @ebullient - Oops sorry  :blush: Didn't mean to commit that. Commit undone. The joys of working on different docs with different rulesets!

**Review by ebullient — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks @michelle-purcell !

**Pull request merged**